lightning/src/chain/channelmonitor.rs

```diff
@@ -1813,8 +1833,23 @@ impl<ChanSigner: ChannelKeys> ChannelMonitor<ChanSigner> {
 	fn spends_watched_output(&self, tx: &Transaction) -> bool {
 		for input in tx.input.iter() {
 			if let Some(outputs) = self.get_outputs_to_watch().get(&input.previous_output.txid) {
-				for (idx, _script_pubkey) in outputs.iter().enumerate() {
-					if idx == input.previous_output.vout as usize {
+				for (idx, _script_pubkey) in outputs.iter() {
+					if *idx == input.previous_output.vout {
+						#[cfg(test)]
+						{
+							// If the witness is empty this transaction is a dummy one expressely
```

Comment:
Can we just...drop that test and panic instead? What was the rationale behind connecting garbage that should only ever be an indication the user is being duped by a bogus chain source (which is explicitly not in our threat model, at least not yet).

Reply:
Tested was added in #333, but can't find the rational. If I remember loosely, at some point we had bug in our dust HTLC canceling back logic at commitment transaction confirmation.  Mutating with the following doesn't break the test so I presume it was an oversight as it doesn't _actually_ cover anything. Removed.

Diff:
```
diff --git a/lightning/src/chain/channelmonitor.rs b/lightning/src/chain/channelmonitor.rs
index 3af98121..927d9d70 100644
--- a/lightning/src/chain/channelmonitor.rs
+++ b/lightning/src/chain/channelmonitor.rs
@@ -1418,12 +1418,12 @@ impl<ChanSigner: ChannelKeys> ChannelMonitor<ChanSigner> {
                                        }
                                }
                        }
-                       if let Some(ref txid) = self.current_counterparty_commitment_txid {
-                               check_htlc_fails!(txid, "current", 'current_loop);
-                       }
-                       if let Some(ref txid) = self.prev_counterparty_commitment_txid {
-                               check_htlc_fails!(txid, "previous", 'prev_loop);
-                       }
+                       //if let Some(ref txid) = self.current_counterparty_commitment_txid {
+                       //      check_htlc_fails!(txid, "current", 'current_loop);
+                       //}
+                       //if let Some(ref txid) = self.prev_counterparty_commitment_txid {
+                       //      check_htlc_fails!(txid, "previous", 'prev_loop);
+                       //}
 
                        if let Some(revocation_points) = self.their_cur_revocation_points {
                                let revocation_point_option =
```

Reply:
I think this test was added in case of future _changes_ of the monitoring code (`check_spend_counterparty`) which may have broken the no-dust-HTLC-canceling-back.